test/integration/negative-rebase.ts

```diff
@@ -0,0 +1,134 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
+
+import { HardhatEthersSigner } from "@nomicfoundation/hardhat-ethers/signers";
+import { setBalance } from "@nomicfoundation/hardhat-network-helpers";
+
+import { ether, impersonate } from "lib";
+import { getProtocolContext, ProtocolContext } from "lib/protocol";
+import { report } from "lib/protocol/helpers/accounting";
+import { norEnsureOperators } from "lib/protocol/helpers/nor";
+import { finalizeWithdrawalQueue } from "lib/protocol/helpers/withdrawal";
+
+import { Snapshot } from "test/suite";
+
+describe("Negative rebase", () => {
+  let ctx: ProtocolContext;
+  let snapshot: string;
+  let beforeTestSnapshot: string;
+  let ethHolder, stEthHolder: HardhatEthersSigner;
+
+  before(async () => {
+    beforeTestSnapshot = await Snapshot.take();
+  });
+
+  after(async () => {
+    await Snapshot.restore(beforeTestSnapshot);
+  });
+
+  beforeEach(async () => {
+    ctx = await getProtocolContext();
+
+    [ethHolder, stEthHolder] = await ethers.getSigners();
+    await setBalance(ethHolder.address, ether("1000000"));
+    const network = await ethers.provider.getNetwork();
+    console.log("network", network.name);
+    if (network.name == "sepolia" || network.name == "sepolia-fork") {
+      const sepoliaDepositContractAddress = "0x7f02C3E3c98b133055B8B348B2Ac625669Ed295D";
+      const bepoliaWhaleHolder = "0xf97e180c050e5Ab072211Ad2C213Eb5AEE4DF134";
+      const BEPOLIA_TO_TRANSFER = 20;
+
+      const bepoliaToken = await ethers.getContractAt("ISepoliaDepositContract", sepoliaDepositContractAddress);
+      const bepiloaSigner = await ethers.getImpersonatedSigner(bepoliaWhaleHolder);
+
+      const adapterAddr = await ctx.contracts.stakingRouter.DEPOSIT_CONTRACT();
+      await bepoliaToken.connect(bepiloaSigner).transfer(adapterAddr, BEPOLIA_TO_TRANSFER);
+    }
+    const beaconStat = await ctx.contracts.lido.getBeaconStat();
+    if (beaconStat.beaconValidators == 0n) {
+      const MAX_DEPOSIT = 150n;
+      const CURATED_MODULE_ID = 1n;
+      const ZERO_HASH = new Uint8Array(32).fill(0);
+      const { lido, depositSecurityModule } = ctx.contracts;
+
+      await finalizeWithdrawalQueue(ctx, stEthHolder, ethHolder);
+
+      await norEnsureOperators(ctx, 3n, 5n);
+
+      const dsmSigner = await impersonate(depositSecurityModule.address, ether("100"));
+      await lido.connect(dsmSigner).deposit(MAX_DEPOSIT, CURATED_MODULE_ID, ZERO_HASH);
+
+      await report(ctx, {
+        clDiff: ether("32") * 3n,
+        clAppearedValidators: 3n,
+        excludeVaultsBalances: true,
+      });
+    }
```

Comment:
This may be moved to the `before` block. Appended after the `beforeTestSnapshot = await Snapshot.take()` (line 22), so it will be run only once before the tests and then reset on beforeEach / afterEach. Wdyt?